registry/storage/cache/metrics/prom.go

```diff
@@ -0,0 +1,69 @@
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/docker/distribution"
+	prometheus "github.com/docker/distribution/metrics"
+	"github.com/docker/distribution/registry/storage/cache"
+	"github.com/docker/go-metrics"
+	"github.com/opencontainers/go-digest"
+)
+
+type prometheusCacheProvider struct {
+	cache.BlobDescriptorCacheProvider
+	latencyTimer metrics.LabeledTimer
+}
+
+func NewPrometheusCacheProvider(wrap cache.BlobDescriptorCacheProvider, name, help string) cache.BlobDescriptorCacheProvider {
+	return &prometheusCacheProvider{
+		wrap,
+		// TODO: May want to have fine grained buckets since redis calls are generally <1ms and the default minimum bucket is 5ms.
```

Comment:
This is something that is required. This will require update in go-metrics API to take in the buckets.

Reply:
Yeah, it would require exposing the prometheus opt through go-metrics. The default bucket is reasonable, this is a good TODO but not a requirement for this.